pkg/loader/compose/compose.go

```diff
@@ -273,12 +322,126 @@ func loadPorts(composePorts []string) ([]kobject.Ports, error) {
 	return ports, nil
 }
 
-// LoadFile loads compose file into KomposeObject
+// LoadFile loads a compose file into KomposeObject
 func (c *Compose) LoadFile(files []string) (kobject.KomposeObject, error) {
+
+	// What will finally be returned
 	komposeObject := kobject.KomposeObject{
 		ServiceConfigs: make(map[string]kobject.ServiceConfig),
 		LoadedFrom:     "compose",
 	}
+
+	// Load the json / yaml file in order to get the version value
+	version, err := getVersionFromFile(files[0])
+	if err != nil {
+		return kobject.KomposeObject{}, errors.Wrap(err, "Unable to load yaml/json file for version parsing")
+	}
+	log.Debugf("Docker Compose version: %s", version)
+
+	// Convert based on version
+	switch version {
+	// Use libcompose for 1 or 2
+	// If blank, it's assumed it's 1 or 2
+	case "", "1", "1.0", "2", "2.0":
+		komposeObject, err := parseV1V2ComposeWithLibcompose(files)
+		if err != nil {
+			return komposeObject, err
```

Comment:
Ah! But the general idea is that, let's say, we are returning `(string, int, error)`, then we return the error but for the other types, we return their zero values, so, `"", 0, err`. Following that, it seems more idiomatic to me to return the zero value for `kobject.KomposeObject` instead of what the function returns.